staging/vhost-device-gpu/src/main.rs

```diff
@@ -67,14 +135,33 @@ pub mod gnu_main {
 
         backend.set_epoll_handler(&daemon.get_epoll_handlers());
 
-        daemon.serve(socket).map_err(Error::ServeFailed)?;
+        daemon.serve(socket_path).map_err(Error::ServeFailed)?;
         Ok(())
     }
 
     pub fn main() {
         env_logger::init();
 
-        if let Err(e) = start_backend(&GpuConfig::from(GpuArgs::parse())) {
+        let args = GpuArgs::parse();
+
+        let flags = GpuFlags {
+            use_gles: args.use_gles,
+            use_egl: args.use_egl,
+            use_glx: args.use_glx,
+            use_surfaceless: args.use_surfaceless,
+        };
```

Comment:
and with the change above this would be:
```suggestion
let flags = GpuFlags::from(args.flags);